hera_qm/utils.py

```diff
@@ -610,3 +613,66 @@ def strip_extension(path, return_ext=False):
         return (root, ext[1:])
     else:
         return os.path.splitext(path)[0]
+
+
+def apply_yaml_freq_time_flags(uv, a_priori_flag_yaml, lat_lon_alt_degrees=None, telescope_name='HERA'):
+    """Apply frequency and time flags to a UVData or UVCal object
+
+    This function takes in a uvdata or uvcal object and applies
+    frequency and time flags as appropriate.
+
+    Parameters
+    ----------
+    uv : UVData or UVCal object
+        uvdata or uvcal to apply frequency / time flags to.
+    a_priori_flag_yaml : str
+        path to yaml file with frequeny / time flags.
+    lat_lon_alt_degrees : list, optional
+        list of latitude, longitude, and altitude for telescope.
+    telescope_name : str, optional
+        string with name of telescope.
+        Default is HERA
+
+    Returns
+    -------
+        uv : UVData or UVCal object
+            input uvdata / uvcal but now with flags applied
+    """
+    # if UVCal provided, get lst_array from times.
+    # If lat_lon_alt is not specified, try to infer it from the telescope name, which calfits files generally carry around
+    if issubclass(uv.__class__, UVCal):
+        if lat_lon_alt_degrees is None:
+            if telescope_name.upper() == 'HERA':
+                lat_lon_alt_degrees = np.array(uvutils.LatLonAlt_from_XYZ(HERA_TELESCOPE_LOCATION))
+                lat_lon_alt_degrees *= [180 / np.pi, 180 / np.pi, 1]
+            else:
+                raise NotImplementedError(f'No known position for telescope {telescope_name}. lat_lon_alt_degrees must be specified.')
+
+        # calculate LST grid in hours from time grid and lat_lon_alt
+        lst_array = uvutils.get_lst_for_time(uv.time_array, *lat_lon_alt_degrees) * 12 / np.pi
+    elif issubclass(uv.__class__, UVData):
+        lst_array = np.unique(uv.lst_array) * 12  / np.pi
+    else:
+        raise NotImplementedError("uv must be a UVData or UVCal object.")
+    time_array = np.unique(uv.time_array)
+    # loop over spws to apply frequency flags.
+    for spw in range(uv.Nspws):
+        flagged_channels = metrics_io.read_a_priori_chan_flags(a_priori_flag_yaml, freqs=uv.freq_array[spw])
+        flagged_channels = flagged_channels[(flagged_channels>=0) & (flagged_channels<=uv.Nfreqs)]
```

Comment:
Should we put this check into `read_a_priori_chan_flags`? Perhaps with a warning if flags are being dropped?

Reply:
Certainly we can put the channel flags >= 0 pretty easily by having it error if they are < 0

Reply:
I don't think this check can always be performed in `read_a_priori_chan_flags` since frequencies are not necessarily provided so it will be tricky too determine if the channel flags exceed maximum channels. I will add a warning here though that flags are being dropped in this method though.